lib/dal/src/cached_module.rs

```diff
@@ -386,6 +406,154 @@ impl CachedModule {
         Ok(result)
     }
 
+    pub async fn get_user_scoped_modules(
+        ctx: &DalContext,
+        user_pk: &UserPk,
+    ) -> CachedModuleResult<Vec<CachedModule>> {
+        let query = "
+            SELECT DISTINCT ON (schema_id)
+                id,
+                schema_id,
+                schema_name,
+                display_name,
+                category,
+                link,
+                color,
+                description,
+                component_type,
+                latest_hash,
+                created_at,
+                NULL::bytea AS package_data,
+                scoped_to_user_pk
+            FROM cached_modules
+            WHERE scoped_to_user_pk = $1
+            ORDER BY schema_id, created_at DESC
+        ";
+
+        let rows = ctx.txns().await?.pg().query(query, &[user_pk]).await?;
+
+        let mut result = vec![];
+
+        for row in rows {
+            result.push(row.try_into()?);
+        }
+
+        Ok(result)
+    }
+
+    // TO-DO: Paul
+    // I will refactor this + insert together - this is to temporarily get something working
+    async fn insert_private_module(
```

Comment:
I have done this on purpose for now - I am testing the viability of this and don't want to break any of the existing routes rigth now - I will refactor this + insert into a single function